core/src/main/scala/org/apache/spark/sql/hive/HiveClientUtil.scala

```diff
@@ -148,12 +148,21 @@ class HiveClientUtil(val sparkContext: SparkContext) extends Logging {
 
     val (useSnappyStore, dbURL, dbDriver) = resolveMetaStoreDBProps()
     if (useSnappyStore) {
-      logInfo(s"Using SnappyStore as metastore database, dbURL = $dbURL")
-      metadataConf.setVar(HiveConf.ConfVars.METASTORECONNECTURLKEY, dbURL)
-      metadataConf.setVar(HiveConf.ConfVars.METASTORE_CONNECTION_DRIVER,
-        dbDriver)
-      metadataConf.setVar(HiveConf.ConfVars.METASTORE_CONNECTION_USER_NAME,
-        Misc.SNAPPY_HIVE_METASTORE)
+      val user = sparkConf.getOption("snappydata.store.user")
+      val password = sparkConf.getOption("snappydata.store.password")
+      val secureDbURL = if (user.isDefined && password.isDefined) {
+        metadataConf.setVar(HiveConf.ConfVars.METASTORE_CONNECTION_USER_NAME, user.get)
+        metadataConf.setVar(HiveConf.ConfVars.METASTOREPWD, password.get)
+        dbURL +";user=" + user.get + ";password=" + password.get + ";default-schema=" +
+            Misc.SNAPPY_HIVE_METASTORE + ";"
+      } else {
+        metadataConf.setVar(HiveConf.ConfVars.METASTORE_CONNECTION_USER_NAME,
+          Misc.SNAPPY_HIVE_METASTORE)
+        dbURL
+      }
+      logInfo(s"Using SnappyStore as metastore database, dbURL = $secureDbURL")
+      metadataConf.setVar(HiveConf.ConfVars.METASTORECONNECTURLKEY, secureDbURL)
+      metadataConf.setVar(HiveConf.ConfVars.METASTORE_CONNECTION_DRIVER, dbDriver)
```

Comment:
They take properties from different places so a common method is not implemented.